pages/admin/treatments/index.tsx

```diff
@@ -0,0 +1,283 @@
+import { Box, Button, Text, useBreakpointValue } from '@chakra-ui/react';
+import { Tooltip } from '@/components/ui/tooltip';
+import React, { useState } from 'react';
+import AdminTable from '../AdminTable';
+import { useTreatments } from '@/hooks/useTreatments';
+import { Pencil, Plus, Trash } from 'lucide-react';
+import TreatmentModal, {
+    ModalMode,
+    TreatmentModalEvent,
+} from './components/TreatmentModal';
+import { Treatment } from '@/interfaces/treatment/Treatment';
+import {
+    createTreatment,
+    deleteTreatment,
+    updateTreatment,
+} from '@/services/TreatmentService';
+import { Toaster } from '@/components/ui/toaster';
+import {
+    hasTreatmentChanges,
+    showNoChangesToast,
+} from '@/utils/treatments/treatmentValidation';
+import TreatmentNotification from '@/utils/notifications';
+import { useFilters } from '@/hooks/useFilters';
+import { SearchFilters } from '@/components/SearchFilters';
+
+export default function TreatmentsPage() {
+    // Loading only for treatments
+    const isSmallScreen = useBreakpointValue({ base: true, md: false });
+    const [isSubmitting, setIsSubmitting] = useState(false);
+    const [modalTrigger, setModalTrigger] = useState<{
```

Comment:
💅🏽 not sure, but I think this sounds more like a modalAction rather than a modalTrigger 👀 

```suggestion
    const [modalAction, setModalAction] = useState<{
```